test/README.md

```diff
@@ -0,0 +1,103 @@
+which would have been created by the testing environment with a running iRODS server.
+
+The docker network used will be referred to as `<NETWORK>` in the following sections.
+
+## Startup the Keycloak image for testing
+
+To run tests that require an OpenID Provider, make sure you build the image.
+To build the image simply run the following command in the `keycloak` directory:
+
+```console
+docker build -f irods-http-api-keycloak.Dockerfile -t irods-http-api-keycloak .
+```
+
+After which, run the following command to startup keycloak:
+
+```console
+docker run --rm --network <NETWORK> -p 8080:8080 irods-http-api-keycloak start-dev
+```
+
+### Get detailed network information
+
+Detailed information of the network will be required to properly configure the HTTP API.
+For example, `/http_server/host` will need to be set to the correct address given by the network
+configuration.
+
+To get this information run the following command:
+```console
+docker network inspect <NETWORK>
+```
+
+## Startup the iRODS HTTP API server
+
+See the [iRODS HTTP API README](/README.md) on how to build the iRODS HTTP API runner image. As well as launching the container.
+
+Be sure to add the `--network <NETWORK>` flag, to have the container be able to communicate with the iRODS server, as well as the Keycloak server.
+
+An example of how the command may look is as follows:
+```console
+docker run -d --rm --name irods_http_api \
+    --network <NETWORK> \
+    -v /path/to/config/file:/config.json:ro \
+    -p 9000:9000 \
+    irods-http-api-runner
+```
+
+### Ensure the HTTP API can communicate with the iRODS server and the keycloak image.
+
+If the HTTP API cannot communicate with the Keycloak server, the HTTP API should not
+complete the startup process.
+
+Additionally, if you [get detailed network information](#get-detailed-network-information), you should see all of the previously started containers listed within the network.
+
+## Configure the [config.py](config.py) to the appropriate test configuration
+
+The configuration should reflect that of the iRODS HTTP API server configuration, otherwise, the incorrect tests may be ran.
+For example, having the HTTP API server set to `protected_resource` mode, while leaving the test configuration in `client` mode
+will produce false errors.
+
+## Install `pytest` to run all tests
+
+Simply run the following command:
+```console
+pip install pytest
+```
+
+Afterwards you should be able to run all the tests by running the following command:
+```console
+pytest
+```
+
+# Testing support for OpenID Connect via Keycloak
+
+Provided in the [`keycloak`](keycloak/) directory is a Dockerfile used for testing
+all OpenID related features. This Dockerfile, [irods-http-api-keycloak.Dockerfile](keycloak/irods-http-api-keycloak.Dockerfile),
+depends on [example-realm-export.json](keycloak/example-realm-export.json) to
+provide the realm used for testing.
```

Comment:
This feels like it should be earlier in the document.

If this section was removed, would the instructions allow readers to setup a working testing environment for OIDC? If the answer is yes, consider removing this.

Reply:
If we want some BG on the Keycloak, then we should have it. Otherwise, we should get rid of it.

Do we think we will remember these details in the future?

Reply:
I defer to you on whether it's worth keeping or not. I think adding a line explaining whether this section is important for standing up the test environment is mostly what's needed. If this section isn't required for testing, then we should state that so users know that they can ignore it.

Alternatively, you could move this into a README that lives in the keycloak directory. Then it's out the way and anyone who wants to know the details about the keycloak docker files can look there.

Thoughts?

Reply:
Not 100% sure on having a keycloak dir readme. Feels a bit hidden away, but we can go with it for now and always move things later.

Reply:
Moved.